lib/bson/regexp.rb

```diff
@@ -156,9 +156,9 @@ def compile
       # @since 3.0.0
       def initialize(pattern, options = '')
         if pattern.include? NULL_BYTE
-          raise Error::InvalidRegexpPattern, "Regexp pattern cannot contain a null byte: #{key}"
-        elsif options.is_a?(String) && options.include?(NULL_BYTE)
-          raise Error::InvalidRegexpPattern, "Regexp options cannot contain a null byte: #{key}"
+          raise Error::InvalidRegexpPattern, "Regexp pattern cannot contain a null byte: #{pattern}"
+        elsif options.to_s.include? NULL_BYTE
```

Comment:
Same as in the above hunk, given that options can be an integer:

- if `options` is a string, use `include?`
- if `options` is a symbol, use `to_s.include?`
- if `options` is an integer, do nothing
- otherwise raise an exception that the options are of an invalid type

Reply:
done